pdfly/cat.py

```diff
@@ -87,8 +91,15 @@ def main(
                     f"WARNING: Page range {page_range} is out of bounds",
                     file=sys.stderr,
                 )
-            for page_num in range(*page_range.indices(len(reader.pages))):
-                writer.add_page(reader.pages[page_num])
+            if not use_complements:
+                for page_num in range(*page_range.indices(len(reader.pages))):
+                    writer.add_page(reader.pages[page_num])
+            else:
+                all_page_nums = set(range(len(reader.pages)))
+                page_nums = set(range(*page_range.indices(len(reader.pages))))
+                compl_page_nums = all_page_nums - page_nums
+                for page_num in compl_page_nums:
+                    writer.add_page(reader.pages[page_num])
```

Comment:
I'm not sure that this is the best place in the code to put that logic...

Wouldn't it be clearer to have the page range inversion made in `parse_filepaths_and_pagerange_arg()`?

On line 62 of this function, we could have:
```python
filename_page_ranges = parse_filepaths_and_pagerange_args(
    filename, fn_pgrgs, inverted_page_selection
)
```

Reply:
The reason I put this logic is here is because the reader objects have been created for each file and then it is possible to use them to know which are the pages that contain each pdf. However, if you think is cleaner and still efficient to create the reader objects again inside the function parse_filepaths_and_pagerange_args to have this logic there, it could be a better solution.

Reply:
Let me know your opinion about this

Reply:
Oh, OK, I understand now that it cannot easily be moved into the `parse_filepaths_and_pagerange_args()` function.

It's alright then